File: Homework/Ngan_Luu/283-move-zeros.py
# https://leetcode.com/problems/move-zeroes/
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        p = 0
        for i in range(1,len(nums)):
            if nums[p]!=0:
                p+=1
            elif nums[i]!=0:
                nums[i],nums[p]=nums[p],nums[i]
                p+=1

File: Homework/Duc_Trinh/lesson-1/283-move-zeros.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
  let correctIndex = 0;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] !== 0) {
      nums[correctIndex] = nums[i];
      correctIndex = correctIndex + 1;
    }
  }
  for (let i = correctIndex; i < nums.length; i++) {
    nums[i] = 0;
  }
};
